mrrc/util.py

```diff
@@ -1,8 +1,59 @@
+from enum import Enum
 import os
+import hashlib
+import errno
+
+class HashType(Enum):
+    """Possible types of hash"""
+    MD5 = 0
+    SHA1 = 1
+    SHA256 = 2
 
 def write_file(file_path:str, content:str):
     if not os.path.isfile(file_path):
         with open(file_path, mode='a'): 
             pass
     with open(file_path, mode='w') as f:
-        f.write(content)
\ No newline at end of file
+        f.write(content)
+        
+def read_sha1(file: str) -> str:
+    """ This function will read sha1 hash of a file from a ${file}.sha1 file first, which should contain
+        the sha1 has of the file. This is a maven repository rule which contains .sha1 files for artifact
+        files. We can use this to avoid the digestion of big files which will improve performance.
+        BTW, for some files like .md5, .sha1 and .sha256, they don't have .sha1 files as they are used for 
+        hashsing, so we will directly calculate its sha1 hash through digesting.
+    """
+    if os.path.isfile(file):
+        non_search_suffix=[".md5", ".sha1", ".sha256"]
```

Comment:
Hi, just a little concern there, this is just for maven upload case and it seems that no metadata file from the maven input's unzip contents, so don't need to consider excluding the metadata ones so far, right?

Reply:
These files are considered to ignore reading sha1 from .sha1 file as its confirmed the .sha1 file does not exist for these files. So its sha1 content is reading directly through digest from the files themselves. This will not break any following logic. You can see the code below for how it is working.

Reply:
Ok, I see this could also apply well for npm on checksum digest. Just make me thinking about metadata files, like maven-metadata.xml or package.json, do we also need to find then compare the checksum(+rh-product) metadata for them? Perhaps I comment in a wrong code line here, sorry for this.

Reply:
Um... you're right, we should ignore some metadata files checking in s3 client. I'll add the logic.

Reply:
@yma88 I've added a new function in s3client to handle the metadata files separately.